chocolatey/choco
# Don't attempt to iterated hook scripts if they're empty/null

## Description Of Changes

Update `chocolateyScriptRunner` to check if any hook scripts were passed in before trying to execute them.

## Motivation and Context

PowerShell 2 had the behaviour of trying to iterate over a null variable. This results in failures when run on PowerShell version 2.

## Testing

1. Run tests in Vagrant before making the change to `chocolateyScriptRunner.ps1`
2. Run tests in Vagrant after making the change to `chocolateyScriptRunner.ps1`
3. Note that tests fail before the change, and succeed after the change.
4. Run a few manual tests on Windows 7 with PowerShell 2

### Operating Systems Testing

- Windows 7 with PowerShell 2
- Windows Server 2016
- Windows Server 2019
- Windows Server 2022

## Change Types Made


## Change Checklist


## Related Issue

Fixes #3088
